### ISSUE TITLE
Feature Request: Support Relation Enforcement (speed camera etc..)

### ISSUE THREAD

**Issue by bitboy85** [organicmaps/organicmaps#2693]
Please support this tag: https://wiki.openstreetmap.org/wiki/Relation:enforcement
Having camera warnings enabled, this will prevent wrong announcements if you drive in opposite direction.

**Comment by biodranik** [comment]
Isn't it already supported?

**Comment by bitboy85** [comment]
Seems not. If camera warnings are enabled, the warning should appear at the place where the relation starts, i guess. This does not happen.

**Comment by MisterY** [comment]
> Seems not. If camera warnings are enabled, the warning should appear at the place where the relation starts, i guess. This does not happen.

You mean, the warning should appear **before** the enforcement starts? :)

**Comment by bitboy85** [comment]
> > Seems not. If camera warnings are enabled, the warning should appear at the place where the relation starts, i guess. This does not happen.
> 
> You mean, the warning should appear **before** the enforcement starts? :)

Yes and No :) It depends how it is mapped. The description in the wiki says for example the "from"-node should start where a sign like "radar control" exists. In reality this is often wayyyyyy before the actual controlling device.

I guess the best option would be to warn before the device if driving in the direction of the enforcement. And ignore it if driving in the opposite direction.

**Comment by MisterY** [comment]
> I guess the best option would be to warn before the device if driving in the direction of the enforcement. And ignore it if driving in the opposite direction.

Agree. I have been checking some cameras that don't get reported (see the linked issue) and noticed that the enforcement is usually (so far it's been always) marked in the area actually covered by the camera. That would mean that it is way too late for any warnings at that point.

**Comment by biodranik** [comment]
Concrete average-speed case in Spain: at 42.653323, -5.596120 a user driving south reports no advance warning; OM warns only about 300 m after passing the device. OSM currently has two average_speed enforcement relations for opposite directions (19573383 and 19573384), so this is a useful bidirectional relation test.
